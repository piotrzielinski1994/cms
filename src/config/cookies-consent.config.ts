const CookiesConsentConstants = {
  STORAGE_KEY: 'cookies-consent',
} as const;

export { CookiesConsentConstants };
