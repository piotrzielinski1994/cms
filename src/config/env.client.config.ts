import { z } from 'zod';

const envSchema = z.object({
  // Default
  NODE_ENV: z.enum(['test', 'development', 'production']),
  NEXT_RUNTIME: z.enum(['nodejs', 'edge']).optional(),
  NEXT_PUBLIC_SITE_NAME: z.string(),
  NEXT_PUBLIC_SERVER_URL: z.string().url(),
  NEXT_PUBLIC_STORYBOOK_URL: z.string().url(),
  // Other
  NEXT_PUBLIC_SENTRY_DSN: z.string().url().optional(),
  NEXT_PUBLIC_GTM_ID: z
    .string()
    .regex(/^GTM-[A-Z0-9]+$/)
    .optional(),
});

const parsedEnv = envSchema.parse({
  NODE_ENV: process.env.NODE_ENV,
  NEXT_RUNTIME: process.env.NEXT_RUNTIME || undefined,
  NEXT_PUBLIC_SITE_NAME: process.env.NEXT_PUBLIC_SITE_NAME,
  NEXT_PUBLIC_SERVER_URL: process.env.NEXT_PUBLIC_SERVER_URL,
  NEXT_PUBLIC_STORYBOOK_URL: process.env.NEXT_PUBLIC_STORYBOOK_URL,
  NEXT_PUBLIC_SENTRY_DSN: process.env.NEXT_PUBLIC_SENTRY_DSN || undefined,
  NEXT_PUBLIC_GTM_ID: process.env.NEXT_PUBLIC_GTM_ID || undefined,
});

const clientEnv = {
  siteName: parsedEnv.NEXT_PUBLIC_SITE_NAME,
  publicUrl: parsedEnv.NEXT_PUBLIC_SERVER_URL,
  storybookUrl: parsedEnv.NEXT_PUBLIC_STORYBOOK_URL,
  sentryDsn: parsedEnv.NEXT_PUBLIC_SENTRY_DSN,
  gtmId: parsedEnv.NEXT_PUBLIC_GTM_ID,
  internal: {
    nodeEnv: parsedEnv.NODE_ENV,
    nextRuntime: parsedEnv.NEXT_RUNTIME,
  },
};

export { clientEnv };
