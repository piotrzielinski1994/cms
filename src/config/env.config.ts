import { z } from 'zod';

const envSchema = z.object({
  // Default
  PAYLOAD_SECRET: z.string().min(10),
  DATABASE_URI: z.string(),
  NEXT_PUBLIC_SERVER_URL: z.string().url(),

  // Vercel
  BLOB_READ_WRITE_TOKEN: z.string().optional(),
});

const parsedEnv = envSchema.parse({
  PAYLOAD_SECRET: process.env.PAYLOAD_SECRET,
  DATABASE_URI: process.env.DATABASE_URI,
  NEXT_PUBLIC_SERVER_URL: process.env.NEXT_PUBLIC_SERVER_URL,
  VERCEL_PROJECT_PRODUCTION_URL: process.env.VERCEL_PROJECT_PRODUCTION_URL,
  BLOB_READ_WRITE_TOKEN: process.env.BLOB_READ_WRITE_TOKEN,
});

const env = {
  payloadSecret: parsedEnv.PAYLOAD_SECRET,
  dbUri: parsedEnv.DATABASE_URI,
  publicUrl: parsedEnv.NEXT_PUBLIC_SERVER_URL,
  vercel: {
    blobStorageToken: parsedEnv.BLOB_READ_WRITE_TOKEN,
  },
};

export { env };
