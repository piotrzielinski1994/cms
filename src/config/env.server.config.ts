import { z } from 'zod';

const envSchema = z.object({
  // Default
  PAYLOAD_SECRET: z.string().min(10),
  DATABASE_URI: z.string(),

  // Other
  SENTRY_ORG: z.string().optional(),
  SENTRY_PROJECT: z.string().optional(),

  // Vercel
  BLOB_READ_WRITE_TOKEN: z.string().optional(),
});

const parsedEnv = envSchema.parse({
  PAYLOAD_SECRET: process.env.PAYLOAD_SECRET,
  DATABASE_URI: process.env.DATABASE_URI,
  SENTRY_ORG: process.env.SENTRY_ORG,
  SENTRY_PROJECT: process.env.SENTRY_PROJECT,
  BLOB_READ_WRITE_TOKEN: process.env.BLOB_READ_WRITE_TOKEN,
});

const serverEnv = {
  payloadSecret: parsedEnv.PAYLOAD_SECRET,
  dbUri: parsedEnv.DATABASE_URI,
  sentry: {
    org: parsedEnv.SENTRY_ORG,
    project: parsedEnv.SENTRY_PROJECT,
  },
  vercel: {
    blobStorageToken: parsedEnv.BLOB_READ_WRITE_TOKEN,
  },
};

export { serverEnv };
