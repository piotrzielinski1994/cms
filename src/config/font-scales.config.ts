import { fullTailwindConfig } from '@/tailwind-config';

const fontScales = {
  xs: 12,
  base: 16,
  xl: 20,
} satisfies Partial<Record<keyof typeof fullTailwindConfig.theme.fontSize, number>>;

export { fontScales };
