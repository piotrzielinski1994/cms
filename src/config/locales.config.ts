import { en as customEn } from '@/payload/locale/en';
import { pl as customPl } from '@/payload/locale/pl';
import {
  DefaultTranslationKeys,
  NestedKeysStripped,
  SupportedLanguages,
  TFunction,
} from '@payloadcms/translations';
import { en } from '@payloadcms/translations/languages/en';
import { pl } from '@payloadcms/translations/languages/pl';
import { pick } from 'ramda';

// Types ====================================

type AdminTranslations = TFunction<NestedKeysStripped<typeof customEn> | DefaultTranslationKeys>;

// Variables ====================================

const contentLocales = ['en', 'pl'] satisfies (keyof SupportedLanguages)[];
const defaultContentLocale = 'en' satisfies (typeof contentLocales)[number];

const adminLocales = pick(contentLocales, { en, pl }) satisfies SupportedLanguages;
const defaultAdminLocale = 'en' satisfies keyof typeof adminLocales;

const customTranslations = {
  en: customEn,
  pl: customPl,
} satisfies Record<(typeof contentLocales)[number], typeof customEn>;

export {
  adminLocales,
  contentLocales,
  customTranslations,
  defaultAdminLocale,
  defaultContentLocale,
  type AdminTranslations,
};
