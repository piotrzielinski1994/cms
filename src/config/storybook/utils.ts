import { ReactNode } from 'react';

const DEFAULT_VALUE = '__default__' as const;
const THUMBNAIL_ID = 'thumbnail';

const storyToUrlPath = (title: string) => {
  const parts = title.split('/');
  const story = parts[parts.length - 1];
  const titlePart = title.toLowerCase().replace(/\//g, '-').replace(/\s+/g, '-');
  const storyPart = story
    .replace(/([a-z0-9])([A-Z])/g, '$1-$2')
    .replace(/([a-zA-Z])(\d+)/g, '$1-$2')
    .toLowerCase();
  return `${titlePart}--${storyPart}`;
};

const getFallback = <T extends ReactNode>(value: T, translation: string): T => {
  return value === DEFAULT_VALUE ? (translation as T) : value;
};

export { DEFAULT_VALUE, getFallback, storyToUrlPath, THUMBNAIL_ID };
