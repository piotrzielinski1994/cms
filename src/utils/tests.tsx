import { Providers } from '@/providers';
import { ComponentProps, ReactNode } from 'react';

const resolveServerComponent = async <T extends (props: unknown) => Promise<ReactNode>>(
  Component: T,
  props: Parameters<T>[0],
) => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};

const withProviders = (
  children: ReactNode,
  props: Partial<Omit<ComponentProps<typeof Providers>, 'children'>> = {},
) => {
  const overwrittenProps: ComponentProps<typeof Providers> = {
    locale: 'en',
    initialTheme: 'light',
    initialFontScale: 'base',
    initialCookiesConsent: true,
    ...props,
    children,
  };
  return <Providers {...overwrittenProps} />;
};

export { resolveServerComponent, withProviders };
