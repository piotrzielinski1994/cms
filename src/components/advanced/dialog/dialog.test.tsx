import { translations } from '@/config/store/locales.config';
import { withProviders } from '@/utils/tests';
import { fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ComponentProps } from 'react';
import { describe, expect, it, vi } from 'vitest';
import { axe } from 'vitest-axe';
import Dialog from './dialog';

describe('Dialog', () => {
  const t = translations.en;
  const footerProps = {
    submitBtn: { label: t.storybook.advanced.dialog.submit, onClick: vi.fn() },
    cancelBtn: { label: t.storybook.advanced.dialog.cancel, onClick: vi.fn() },
  } satisfies ComponentProps<typeof Dialog.Footer>;
  const defaultProps = {
    header: 'Header',
    footer: <Dialog.Footer {...footerProps} />,
    onClose: vi.fn(),
    open: true,
  } satisfies ComponentProps<typeof Dialog.Root>;

  beforeEach(() => {
    vi.resetAllMocks();
  });

  it('should have no accessibility violations', async () => {
    const { container } = render(withProviders(<Dialog.Root {...defaultProps} />));
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(withProviders(<Dialog.Root {...defaultProps} />));
    expect(container).toMatchSnapshot();
  });

  describe('Interactions', () => {
    it('should trigger closing when close button is clicked', async () => {
      const { getByRole } = render(withProviders(<Dialog.Root {...defaultProps} />));

      await userEvent.click(getByRole('button', { name: t.frontend.close }));

      expect(defaultProps.onClose).toHaveBeenCalled();
    });

    it('should trigger submission when submit button is clicked', async () => {
      const { getByRole } = render(withProviders(<Dialog.Root {...defaultProps} />));

      await userEvent.click(getByRole('button', { name: footerProps.submitBtn.label }));

      expect(footerProps.submitBtn.onClick).toHaveBeenCalled();
    });

    it('should trigger cancel event when cancel button is clicked', async () => {
      const { getByRole } = render(withProviders(<Dialog.Root {...defaultProps} />));

      await userEvent.click(getByRole('button', { name: footerProps.cancelBtn.label }));

      expect(footerProps.cancelBtn.onClick).toHaveBeenCalled();
    });

    describe('Keyboard', () => {
      it('should trigger closing modal when Escape key is pressed', () => {
        const { getByRole } = render(withProviders(<Dialog.Root {...defaultProps} type="modal" />));
        const modal = getByRole('dialog');

        fireEvent.keyDown(modal, { key: 'Escape' }); // userEvent was not triggering event

        expect(defaultProps.onClose).toHaveBeenCalled();
      });

      it('should not trigger closing dialog when Escape key is pressed', async () => {
        const { getByRole } = render(
          withProviders(<Dialog.Root {...defaultProps} type="dialog" />),
        );
        const dialog = getByRole('dialog');

        fireEvent.keyDown(dialog, { key: 'Escape' });

        expect(defaultProps.onClose).not.toHaveBeenCalled();
      });
    });
  });
});
