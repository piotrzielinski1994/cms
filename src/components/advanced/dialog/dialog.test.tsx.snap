// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog > should match the snapshot 1`] = `
<div>
  <div
    class="px-4 sm:px-6 cms-section fixed inset-0 z-dialog py-4 sm:py-6 hidden has-[:open]:grid pointer-events-none"
  >
    <div
      class="fixed inset-0 z-backdrop bg-foreground/50 dark:bg-foreground/10 backdrop-blur-none"
    />
    <dialog
      class="w-full max-w-screen-2xl mx-auto grid cms-container relative z-dialog overflow-visible pointer-events-auto bg-background text-foreground focus-visible:tw-cms-dialog-outline"
      data-dialog="true"
      open=""
    >
      <header
        class="flex"
      >
        <p
          class="p-4 md:p-6 flex-grow"
        >
          Header
        </p>
        <button
          aria-label="Close"
          class="p-4 md:p-6 focus-visible:tw-cms-outline"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-x"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6 6 18"
            />
            <path
              d="m6 6 12 12"
            />
          </svg>
        </button>
      </header>
      <div
        class="p-4 pt-0 md:p-6 md:pt-0"
      />
      <footer
        class="p-4 pt-0 md:p-6 md:pt-0"
      >
        <div
          class="flex justify-end gap-2 md:gap-4 flex-wrap"
        >
          <button
            class="px-6 py-3 border-2 border-primary hover:border-primary/90 font-semibold whitespace-nowrap disabled:cursor-not-allowed focus-visible:tw-cms-outline text-primary disabled:border-primary/50 disabled:text-primary/50 bg-background [&:not(:disabled)]:hover:bg-foreground/5"
            type="button"
          >
            Cancel
          </button>
          <button
            class="px-6 py-3 border-2 border-primary hover:border-primary/90 font-semibold whitespace-nowrap disabled:cursor-not-allowed focus-visible:tw-cms-outline bg-primary hover:bg-primary/90 text-primary-foreground disabled:border-primary/0 disabled:bg-primary/50"
            type="button"
          >
            Submit
          </button>
        </div>
      </footer>
    </dialog>
  </div>
</div>
`;
